### DISCUSSION TITLE
Checking necessary packages in image

### DISCUSSION THREAD

**Discussion by AlexK09** [aquasecurity/trivy#5176] [Ideas]
### Description

I suggest to add checking necessary packages in images. For example: I want 2 packages (curl and wget) to be installed in image. If these packages are not installed in some image, I suggest to mark it in configauditreport or vulnerabilityreport. It will be really useful feature because there many companies nowadays who develop their own packages for protecting images.

### Target

None

### Scanner

None

**Comment by knqyf263** [comment]
It sounds like out of scope for Trivy as it is not a security issue.

**Reply by AlexK09** [reply]
There are many policies to provide information security in company. I understand that one tool can't and shouldn't implement all security policies but in this case trivy can really help us.  Trivy can collect information about all installed packages in image - it is really cool and important feature. And my proposal is to match these installed packages with those packages which are necessary for users. I understand that it is not a classic vulnerability. But Trivy also finds misconfigurations and I can definitely say that if I haven't necessary packages in my image - it is a misconfiguration and such issue may be marked in configauditreport where trivy marks misconfigurations.

**Reply by knqyf263** [reply]
You can write a small script against the repost Trivy generates.

**Reply by AlexK09** [reply]
I can, but it is really uncomfortable to use it with trivy-operator, which scans pods when they appears. It will become harder to implement it to existing clusters. It could be really cool feature if such script was realised in trivy. It colud be used like this:
trivy image —necessary-packages curl, wget …
Implementing of one tool is easier than two. Also, it will raise the popularity of Trivy, because this tool will become more flexible.
There are many specialists who could it use.
